src/auxiliary/precompile.jl

```diff
@@ -382,10 +382,10 @@ function _precompile_manual_()
     # end
     # end
     @assert Base.precompile(Tuple{typeof(SummaryCallback)})
-    @assert Base.precompile(Tuple{DiscreteCallback{typeof(Trixi.summary_callback),
-                                                   typeof(Trixi.summary_callback),
-                                                   typeof(Trixi.initialize_summary_callback),
-                                                   typeof(SciMLBase.FINALIZE_DEFAULT)}})
+    # @assert Base.precompile(Tuple{DiscreteCallback{typeof(Trixi.summary_callback),
+    #                                                typeof(Trixi.summary_callback),
+    #                                                typeof(Trixi.initialize_summary_callback),
+    #                                                typeof(SciMLBase.FINALIZE_DEFAULT)}})
```

Comment:
What would the errors be if not commented here?

Reply:
I get the following stacktrack
```julia
julia> using Trixi
Precompiling Trixi...
Info Given Trixi was explicitly requested, output will be shown live 
ERROR: LoadError: AssertionError: Base.precompile(Tuple{DiscreteCallback{typeof(Trixi.summary_callback), typeof(Trixi.summary_callback), typeof(Trixi.initialize_summary_callback), typeof(SciMLBase.FINALIZE_DEFAULT)}})
Stacktrace:
 [1] _precompile_manual_()
   @ Trixi ~/.julia/dev/Trixi/src/auxiliary/precompile.jl:385
 [2] top-level scope
   @ ~/.julia/dev/Trixi/src/Trixi.jl:356
 [3] include
   @ ./Base.jl:557 [inlined]
 [4] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
   @ Base ./loading.jl:2806
 [5] top-level scope
   @ stdin:4
in expression starting at /Users/jchan985/.julia/dev/Trixi/src/Trixi.jl:1
in expression starting at stdin:4
```

Reply:
Ok I will try to look into this

Reply:
Much appreciated. I will see if I can figure out the allocation failure...

Reply:
Type signature changed for `DiscreteCallback` see https://github.com/SciML/SciMLBase.jl/commit/59bfc8a4d3d42b7017aa580b7021f39037c9603a

Reply:
Thank you @huiyuxie! So restricting to [<v2.53.0](https://github.com/SciML/SciMLBase.jl/releases/tag/v2.53.0) might fix this?

Reply:
I think changing with upstream would be better - you could make `typeof(nothing)` as the last type

Reply:
Just to check - do you mean the following?

```julia
@assert Base.precompile(Tuple{DiscreteCallback{typeof(Trixi.summary_callback),
                                            typeof(Trixi.summary_callback),
                                            typeof(Trixi.initialize_summary_callback),
                                            typeof(SciMLBase.FINALIZE_DEFAULT),
                                            typeof(nothing)
                                            }})
```

or 

```julia
@assert Base.precompile(Tuple{DiscreteCallback{typeof(Trixi.summary_callback),
                                            typeof(Trixi.summary_callback),
                                            typeof(Trixi.initialize_summary_callback),
                                            typeof(nothing)
                                            }})
```

Reply:
The first one

Reply:
> Is there any reason for not updating?

No, I think this PR is currently using 2.58.1 anyways (for `VectorOfArray` compat) so it has already updated.

Reply:
Thanks!

Reply:
It works!! Thank you again @huiyuxie, you're a lifesaver.